### PR TITLE
Added new perf tests to show off a few more scenarios.

### PR DESCRIPTION
**Pull request by jonfortescue**
New perf tests added to show off server scenario perf increases. Updated tests to remove setup from foreach loop.

### CODE REVIEW

File: tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs [hunk 1]
```diff
@@ -142,21 +142,72 @@ private unsafe static void ByteStarUnmanagedToByte(string text)
         {
 			int length = text.Length;
 			byte[] utf8ByteArray = UtfEncode(text);
```
Comment (KrzysztofCwalina) [inline tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs]: these two lines don't seemed to be indented properly.

File: tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs [hunk 2]
```diff
@@ -709,21 +927,78 @@ private unsafe static void ByteStarUnmanagedToSbyte(string text)
...
+            byte* unmanagedBytePtr;
```
Comment (KrzysztofCwalina) [inline tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs]: the 2 lines above are not indented properly.

**Comment by KrzysztofCwalina** [comment]
looks good.

**Comment by jonfortescue** [comment]
Not sure what was up with the formatting exactly, but I just did a global ctrl-K,D on the document, so everything should be formatted properly now.

**Comment by KrzysztofCwalina** [comment]
looks good!

**Pull request merged**